### PR TITLE
feat: Redesign resources table

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
**Demo:**

https://user-images.githubusercontent.com/3165839/196266312-147f13cb-f949-4072-9699-817c180ca5ad.mov

**Notable changes:**
- The resources are not displayed in a table anymore, but as a list of cards
- The metadata was moved to the card header. By default, it shows the first 4 values and hides the others. You can show or hide them all tho
- The agent info was simplified. The preferred latency is displayed but you can see all the DERP servers in the popover.
- The resources are sorted by the  number of agents so the resources with agents are displayed first

### CODE REVIEW

**Comment by ntimo** [comment]
I really like the design change, the only thing I am not sure about is that the meta data is at the top. Just thinking about a workspace that has like a lot of resources they would all show above the workspace with the coder apps which are kind of the most important at least for me.

I actually prefer the workspace to be at the top, that's also why I hide all of my other resources like docker volumes since they are not really "important". Another question I am asking my self is how does this redesign work with hidden resources?

**Comment by BrunoQuaresma** [comment]
I'm glad you liked it @ntimo 

>  I really like the design change, the only thing I am not sure about is that the meta data is at the top. Just thinking about a workspace that has like a lot of resources they would all show above the agent with the coder apps which are kind of the most important at least for me.

It displays only the first 4 values to avoid pushing the agents too below but you can see all of them by clicking on the toggle button. Do you think it is still an issue?

> I actually prefer the workspace to be at the top

I didn't understand this. By workspace do you mean the resource with agents? If yes, we changed the sort. Now, the resources are sorted by the number of agents.

**Comment by BrunoQuaresma** [comment]
@ntimo 

> Another question I am asking my self is how does this redesign work with hidden resources?

Pretty similar to the previous version.

File: site/src/util/workspace.ts [hunk 1]
```diff
@@ -106,10 +105,9 @@ export const displayWorkspaceBuildDuration = (
 }
 
 export const DisplayAgentStatusLanguage = {
-  loading: "Loading...",
-  connected: "⦿ Connected",
-  connecting: "⦿ Connecting",
-  disconnected: "◍ Disconnected",
+  connected: "Connected",
+  connecting: "Connecting...",
+  disconnected: "Disconnected",
```
Comment (presleyp) [inline site/src/util/workspace.ts:110]: Might be a good time to move these to i18n

File: site/src/components/Tooltips/HelpTooltip/HelpTooltip.tsx [hunk 2]
```diff
@@ -35,6 +35,35 @@ const useHelpTooltip = () => {
   return helpTooltipContext
 }
 
+export const HelpPopover: React.FC<
```
Comment (presleyp) [inline site/src/components/Tooltips/HelpTooltip/HelpTooltip.tsx:38]: Nice

**Review by bpmct — COMMENTED** [review]
Just gave it a shot with my [kubernetes multi service](https://github.com/bpmct/coder-templates/tree/main/kubernetes) template. It looks great and the "copy value" works like a charm!

1. The agents are still changing order randomly. I confirmed this isn't an issue exclusive to this branch, but perhaps we should reopen #2778 if you don't want to tackle in this PR. 
2. The other  tooltips are difficult to dismiss. I did a fix for this in #4562, do we just need to rebase?

Other then that, LGTM.

**Comment by presleyp** [comment]
I'm wondering if people with a lot of resources will not realize the logs exist or not want to scroll down to them. Maybe there could be a way to collapse all resources or skip to the logs, but that can be a future PR (if the concern is even valid, I don't know what the max number of resources will be). 

Also wondering about the accessibility of the version color. To me it's easier to see, but I don't know about to someone color blind or using a screen reader. Maybe the popover addresses the screen reader issue? Not sure how those are handled. Paging @Kira-Pilot 

Other than that, looks great!

**Review by presleyp — APPROVED** [review]

**Pull request merged**